acceptance/testdata/issue/issue-create-edit-with-project.txtar

```diff
@@ -0,0 +1,43 @@
+env REPO=${SCRIPT_NAME}-${RANDOM_STRING}
+
+# Create a repository with a file so it has a default branch
+exec gh repo create ${ORG}/${REPO} --add-readme --private
+
+# Defer repo cleanup
+defer gh repo delete --yes ${ORG}/${REPO}
+
+# Create a project
+env PROJECT_TITLE=${REPO}-project
+exec gh project create --owner=${ORG} --title=${PROJECT_TITLE} --format='json' --jq='.number'
+stdout2env PROJECT_NUMBER
+
+defer gh project delete --owner=${ORG} ${PROJECT_NUMBER}
+
+# Clone the repo
+exec gh repo clone ${ORG}/${REPO}
+
+# Create an issue in the repo
+cd ${REPO}
+exec gh issue create --title 'Feature Request' --body 'Feature Body' --project ${PROJECT_TITLE}
+stdout2env ISSUE_URL
```

Comment:
Revisiting the code here, `gh issue create --project` will try to resolve the provided project title against v1 and v2 projects alike ...

https://github.com/cli/cli/blob/0dd9f4715910cb0ea32e96d05fb9cf7a5da66016/pkg/cmd/pr/shared/params.go#L96-L122

... before creating the issue, which needs to make follow up GraphQL mutation because v2 projects aren't supported in creating issues ...

https://github.com/cli/cli/blob/0dd9f4715910cb0ea32e96d05fb9cf7a5da66016/api/queries_issue.go#L184-L239